src/shell/file_reader.cpp

```diff
@@ -38,9 +38,9 @@ std::optional<uint8_t> FileReader::Read()
 {
 	std::uint8_t data        = 0;
 	std::uint16_t bytes_read = 1;
-	DOS_ReadFile(handle, &data, &bytes_read);
 
-	if (bytes_read == 0) {
+	bool result = DOS_ReadFile(handle, &data, &bytes_read);
+	if (!result || bytes_read == 0) {
```

Comment:
The logic itself looks good.

Presumably if `DOS_ReadFile` gives us a negative return code, we shouldn't trust the data (even if some quantity of bytes were read).